kolibri/core/assets/test/vue.progress-bar.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-env mocha */
+// The following two rules are disabled so that we can use anonymous functions with mocha
+// This allows the test instance to be properly referenced with `this`
+/* eslint prefer-arrow-callback: "off", func-names: "off" */
+
+'use strict';
+
+const Vue = require('vue-test');
+const progressBar = require('../src/vue/progress-bar');
+const ProgressBarComponent = Vue.extend(progressBar);
+const assert = require('assert');
+
+describe('progressBar Component', function () {
+  describe('computed properties', function () {
+    describe('percent', function () {
+      beforeEach(function () {
+        this.vm = new ProgressBarComponent({
+          propsData: {
+            progress: 0,
+          },
+        }).$mount();
+      });
+      it('should give 0 percent for progress of < 0', function () {
```

Comment:
Things that aren't declarative in nature has been my rule of thumb so far. So, testing that the template renders as a template is more about testing that Vue works as we expect it to, which is rather beyond our purview.

So, in this example, I have tested the computed properties of the component, because those are the only things that have some sort of input and output that we want to guarantee. You will note that as a result of the tests here, I added a minimum value as well as the max to prevent weird behaviour if a negative is passed in mistakenly.